### PR TITLE
Add: Featured cards on landing page

### PR DESCRIPTION
**Pull request by LordUttam**
Now the Featured Cards on Landing page take you to products page with category filter applied.

image

### CODE REVIEW

File: src/backend/db/categories.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { v4 as uuid } from "uuid";
-
+import featImg1 from "assets/images/categories/bitsat.jpg";
+import featImg2 from "assets/images/categories/selfhelp.png";
+import featImg3 from "assets/images/categories/classics.jpg";
```
Comment (Mohasin-Haque) [inline src/backend/db/categories.js]: You can make one single file for imports and then import it here

**Review by Mohasin-Haque — COMMENTED** [review]
Your code is good . You can work on the comments and you are good to merge

File: src/components/landing-components/trending.jsx [hunk 2]
```diff
@@ -1,57 +1,23 @@
...
+      setProducts(response.data.products.slice(0, 6));
+    })();
+  }, []);
+
   return (
     <>
       <h2 className="h--2 trending__heading m__t-1 m__b-0">Trending</h2>
       <section className="trending flex justify--center p--x-2 p--y-2">
-        <TrendingCard
-          trendImg={trendImg1}
```
Comment (anshtripathi01) [inline src/components/landing-components/trending.jsx:15]: Why did you passing data like this

File: src/components/landing-components/trending.jsx [hunk 3]
```diff
@@ -1,57 +1,23 @@
...
-          bookTitle="Savarkar (Part 1)"
-          author="Vikram Sampath"
-          price="Rs. 99"
-        />
-
```
Comment (anshtripathi01) [inline src/components/landing-components/trending.jsx:20]: You can make separate file for this

**Review by anshtripathi01 — APPROVED** [review]
Keep above points rest codes and formate looks good to merge

File: src/components/cards/cards.jsx [hunk 4]
```diff
@@ -8,52 +9,96 @@ function FeaturedCard(props) {
...
   const cardSubtitle = props.cardSubtitle;
+  const categoryId = props.categoryId;
+  const navigate = useNavigate();
+  const { dispatch } = useProducts();
+
+  const explore = () => {
+    dispatch({ type: "CLEAR_ALL" });
+    dispatch({ type: "CATEGORIES", payload: categoryId });
+    navigate("/products");
+    console.log(categoryId);
```
Comment (JatuFaique) [inline src/components/cards/cards.jsx]: You can clear console logs.

File: src/components/landing-components/featured.jsx [hunk 5]
```diff
@@ -1,57 +1,31 @@
...
+
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/categories");
+      setFeaturedCategories(response.data.categories);
+    })();
+  }, []);
+
   return (
     <>
```
Comment (JatuFaique) [inline src/components/landing-components/featured.jsx:16]: remove this , a return state only requires a single closing div element.

**Review by JatuFaique — APPROVED** [review]
clean code practices followed
    good folder structure
    good to merge

File: src/components/landing-components/featured.jsx (hunk 5 above)
Comment (LordUttam) [reply src/components/landing-components/featured.jsx:16]: remove what? it is a single element

File: src/components/landing-components/trending.jsx (hunk 2 above)
Comment (LordUttam) [reply src/components/landing-components/trending.jsx:15]: Those lines are in removed, see the minus on left?

**Pull request merged**